bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingFactory.java

```diff
@@ -105,7 +105,7 @@ public static Thing createThing(ThingUID thingUID, Configuration configuration,
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                if (properties != null) {
+                if (thing != null && properties != null) {
```

Comment:
This is an error case - not a standard case. In this case, the handler has already said that it supports the thing type, but it then fails to create it, and returns null. I guess we could loop to see if another factory wants a go, but if we hit this case, something is wrong and IMHO we probably should stop.